centos/centos-8.x/centos-8.3-hpc/install_mellanoxofed.sh

```diff
@@ -1,15 +1,24 @@
 #!/bin/bash
 set -ex
 
-MLNX_OFED_DOWNLOAD_URL=https://azhpcstor.blob.core.windows.net/azhpc-images-store/MLNX_OFED_LINUX-5.2-1.0.4.0-rhel8.3-x86_64.tgz
-TARBALL=$(basename ${MLNX_OFED_DOWNLOAD_URL})
+VERSION="5.8-1.0.1.1"
+$COMMON_DIR/write_component_version.sh "MOFED" $VERSION
+TARBALL="MLNX_OFED_LINUX-$VERSION-rhel8.3-x86_64.tgz"
+MLNX_OFED_DOWNLOAD_URL=https://content.mellanox.com/ofed/MLNX_OFED-${VERSION}/$TARBALL
 MOFED_FOLDER=$(basename ${MLNX_OFED_DOWNLOAD_URL} .tgz)
 
-$COMMON_DIR/download_and_verify.sh $MLNX_OFED_DOWNLOAD_URL "dd13ffa54524af9509e3de7288a863aa48ed7016bb3d15446523703020c41715"
+$COMMON_DIR/download_and_verify.sh $MLNX_OFED_DOWNLOAD_URL "370f49fd9190f046e1a41853e9f03508b57f1a29fa01f73099e564e6529a979d"
 tar zxvf ${TARBALL}
 
 KERNEL=( $(rpm -q kernel | sed 's/kernel\-//g') )
-KERNEL=${KERNEL[-1]}
+# Uncomment the lines below if you are running this on a VM
+# RELEASE=( $(cat /etc/centos-release | awk '{print $4}') )
+# yum install -y https://vault.centos.org/${RELEASE}/BaseOS/x86_64/os/Packages/kernel-devel-${KERNEL}.rpm
 yum install -y kernel-devel-${KERNEL}
-./${MOFED_FOLDER}/mlnxofedinstall --kernel $KERNEL --kernel-sources /usr/src/kernels/${KERNEL} --add-kernel-support --skip-repo
+./${MOFED_FOLDER}/mlnxofedinstall --kernel $KERNEL --kernel-sources /usr/src/kernels/${KERNEL} --add-kernel-support --skip-repo --skip-unsupported-devices-check --without-fw-update
 
+# Issue: Module mlx5_ib belong to a kernel which is not a part of MLNX
+# Resolution: set FORCE=1/ force-restart /etc/init.d/openibd 
+# This causes openibd to ignore the kernel difference but relies on weak-updates
+# Restarting openibd
+/etc/init.d/openibd force-restart
```

Comment:
Check the status after restart?

Reply:
Will add. Thank you.

Reply:
Has the check been added?

Reply:
Yes, please refer - https://github.com/Azure/azhpc-images/blob/b76ab942a14ffdb89f7edf4a0c5268c7cb60e0ad/centos/centos-8.x/centos-8.3-hpc/install_mellanoxofed.sh#:~:text=/etc/init.d/openibd%20force,fi